haskell/cabal
# Filter out files that are not ending in .c from c-sources

This poc pre-filters file paths given in c-sources to only keep the ones ending in ".c", and outputs an appropriate message.

1. Should we check if there are ".h" files explicitly and redirect to "includes"?
2. Should we expand this to cxx-sources?

## QA Notes

1. Clone the repository `https://github.com/jtdaugherty/vty-unix`
2. Remove from its cabal file the exported modules, dependencies and executable component
3. Run `cabal build -w ghc-9.2.8`
4. Make sure the output is the following:

```
Preprocessing library for repro-0.1.0.0..
Building library for repro-0.1.0.0..
Warning: The following files listed in the main library's c-sources will not
be used: cbits/gwinsz.h.
Header files should be in the 'include' or 'install-include' stanza.
See https://cabal.readthedocs.io/en/3.10/cabal-package.html#pkg-field-includes
Preprocessing library 'lib2' for repro-0.1.0.0..
Building library 'lib2' for repro-0.1.0.0..
Warning: The following files listed in library lib2's c-sources will not be
used: cbits/gwinsz.h.
Header files should be in the 'include' or 'install-include' stanza.
See https://cabal.readthedocs.io/en/3.10/cabal-package.html#pkg-field-includes
Preprocessing executable 'exec1' for repro-0.1.0.0..
Building executable 'exec1' for repro-0.1.0.0..
[1 of 1] Compiling Main             ( Main.hs, /home/hecate/Vrac/cabal-9190-repro/dist-newstyle/build/x86_64-linux/ghc-9.4.5/repro-0.1.0.0/x/exec1/build/exec1/exec1-tmp/Main.o )
Warning: The following files listed in exec1's c-sources will not be used:
cbits/gwinsz.h.
Header files should be in the 'include' or 'install-include' stanza.
See https://cabal.readthedocs.io/en/3.10/cabal-package.html#pkg-field-includes
[2 of 2] Linking /home/hecate/Vrac/cabal-9190-repro/dist-newstyle/build/x86_64-linux/ghc-9.4.5/repro-0.1.0.0/x/exec1/build/exec1/exec1
```
5. Exit code should be 0 (you can check with `echo $?`


relates to #9190

@hasufell @mouse07410 @andreabedini @angerman
